libs/libc/stdio/lib_fopencookie.c

```diff
@@ -0,0 +1,156 @@
+/****************************************************************************
+ * libs/libc/stdio/lib_fopencookie.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <string.h>
+#include <assert.h>
+#include <errno.h>
+#include <nuttx/tls.h>
+
+#include "libc.h"
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: cookie_read_cb
+ ****************************************************************************/
+
+static ssize_t cookie_read_cb(FAR void *cookie, FAR char *buf, size_t size)
+{
+  /* Per specification: if *read is a null pointer, then reads from the
+   * custom stream always return end of file.
+   */
+
+  return 0;
+}
+
+/****************************************************************************
+ * Name: cookie_write_cb
+ ****************************************************************************/
+
+static ssize_t cookie_write_cb(FAR void *cookie, FAR const char *buf,
+                               size_t size)
+{
+  /* Per specification: if *write is a null pointer, then output to the
+   * stream is discarded.
+   */
+
+  return 0;
```

Comment:
need return size here

Reply:
We should return no bytes written, no?

Reply:
0 mean nothing is written, but the spec requires discard all input from the comment.

Reply:
Hm, by discarding all input I understand that nothing is written. Returning size would mean the application could expect the pointer is somewhere else then it really is. I will look at other libc implementations how they deal with this.

Reply:
Ok, musl also returns size. Changed.